src/ctl/v1/user.go

```diff
@@ -41,9 +42,15 @@ func createUserHandler(log *xlog.Log, proxy *proxy.Proxy, w rest.ResponseWriter,
 		rest.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	log.Warning("api.v1.create.user[from:%v].[%v]", r.RemoteAddr, p)
 
-	query := fmt.Sprintf("GRANT SELECT ON *.* TO '%s'@'%%' IDENTIFIED BY '%s'", p.User, p.Password)
+	if len(p.Database) == 0 || len(p.User) == 0 || len(p.Password) == 0 {
+		log.Error("api.v1.create.user[%+v].error:some param is empty", p)
+		rest.Error(w, "some param is empty", http.StatusNoContent)
+		return
+	}
+
+	log.Warning("api.v1.create.user[from:%v].[%v]", r.RemoteAddr, p)
+	query := fmt.Sprintf("GRANT ALL ON %s.* TO '%s'@'%%' IDENTIFIED BY '%s'", p.Database, p.User, p.Password)
```

Comment:
It's better to add \`%s\` for the database format.

Reply:
have added it.